The job **`x86_64-gnu-miri`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16940281894/job/48007334665) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/48007334665) [(plain)](https://github.com/rust-lang/rust/commit/7287b93129f1022eeddfd06cdabbff50f86516ec/checks/48007334665/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] rustc_resolve test:false 59.622
   Compiling rustc_monomorphize v0.0.0 (/checkout/compiler/rustc_monomorphize)
[RUSTC-TIMING] rustc_hir_typeck test:false 41.422
   Compiling rustc_query_impl v0.0.0 (/checkout/compiler/rustc_query_impl)
error[E0599]: the method `clone` exists for enum `std::option::Option<CoverageInfoBuilder>`, but its trait bounds were not satisfied
   --> compiler/rustc_mir_build/src/builder/mod.rs:806:52
    |
806 |         body.coverage_info_hi = self.coverage_info.clone().map(|b| b.as_done());
    |                                                    ^^^^^ method cannot be called on `std::option::Option<CoverageInfoBuilder>` due to unsatisfied trait bounds
    |
   ::: compiler/rustc_mir_build/src/builder/coverageinfo.rs:15:1
    |
 15 | pub(crate) struct CoverageInfoBuilder {
    | ------------------------------------- doesn't satisfy `CoverageInfoBuilder: std::clone::Clone`
```

</details>

--- a/compiler/rustc_mir_build/src/builder/coverageinfo.rs
+++ b/compiler/rustc_mir_build/src/builder/coverageinfo.rs
@@ -157,6 +157,21 @@ impl CoverageInfoBuilder {
         // if there's nothing interesting in it.
         Box::new(CoverageInfoHi { num_block_markers, branch_spans })
     }
+
+    pub(crate) fn as_done(&self) -> Box<CoverageInfoHi> {
+        let &Self { nots: _, markers: BlockMarkerGen { num_block_markers }, ref branch_info } =
+            self;
+
+        let branch_spans = branch_info
+            .as_ref()
+            .map(|branch_info| branch_info.branch_spans.as_slice())
+            .unwrap_or_default()
+            .to_owned();
+
+        // For simplicity, always return an info struct (without Option), even
+        // if there's nothing interesting in it.
+        Box::new(CoverageInfoHi { num_block_markers, branch_spans })
+    }
 }
 
 impl<'tcx> Builder<'_, 'tcx> {

--- a/compiler/rustc_mir_build/src/builder/mod.rs
+++ b/compiler/rustc_mir_build/src/builder/mod.rs
@@ -791,7 +791,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
 
     #[allow(dead_code)]
     fn dump_for_debugging(&self) {
-        let body = Body::new(
+        let mut body = Body::new(
             MirSource::item(self.def_id.to_def_id()),
             self.cfg.basic_blocks.clone(),
             self.source_scopes.clone(),
@@ -803,7 +803,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             self.coroutine.clone(),
             None,
         );
-        //body.coverage_info_hi = self.coverage_info.clone().map(|b| b.into_done());
+        body.coverage_info_hi = self.coverage_info.as_ref().map(|b| b.as_done());
 
         use rustc_middle::mir::pretty;
         let options = pretty::PrettyPrintMirOptions::from_cli(self.tcx);

